### ISSUE TITLE
inheritance & the necessity to define base method in derived class?

### ISSUE THREAD

**Issue by DannyDaiSun** [ThePhD/sol2#1204]
@ThePhD Hi, first of all thanks a lot for this great library, so useful. Second, regarding to your message of issue: https://github.com/ThePhD/sol2/issues/214 

> So this bothered me a lot. In the end, I actually ended up working hard on the implementation and enabling inheritance, as well as actual properties / variables. The usual caveats no longer apply: http://sol2.readthedocs.io/en/latest/api/simple_usertype.html
> 
> Give it a try with the latest (v2.14.0), and let me know if it's satisfactory.

does it mean, we no longer need to specify base class member methods when registering derived class via new_usertype<T>? I have tried the following, which works as I expected, but I just worried if there would be a runtime overhead issue?

my code:

    new_usertype<widget>(
        "widget",
        "draw", &widget::draw,
        "rotate", &widget::rotate
        );

    new_usertype<button>(
        "button",
        "draw", &button::draw,
        sol::base_classes, sol::bases<widget>()
        );

context:
    base class: widget
    derived class: button
    virtual function: draw
    base class normal function: rotate

In summary, I wonder if my code is the best way of doing inheritance with sol?

Thank you in advance.

_Originally posted by @DannyDaiSun in https://github.com/ThePhD/sol2/issues/214#issuecomment-849523424_
